apis/kafkamanagement/v1beta1/kafkauser_webhook.go

```diff
@@ -47,3 +47,38 @@ func (r *KafkaUser) Default() {
 		})
 	}
 }
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
+//+kubebuilder:webhook:path=/validate-kafkamanagement-instaclustr-com-v1beta1-kafkauser,mutating=false,failurePolicy=fail,sideEffects=None,groups=kafkamanagement.instaclustr.com,resources=kafkausers,verbs=create;update,versions=v1beta1,name=vkafkauser.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &KafkaUser{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *KafkaUser) ValidateCreate() error {
+	kafkauserlog.Info("validate create", "name", r.Name)
+
+	// TODO(user): fill in your validation logic upon object creation.
+	return nil
+}
```

Comment:
Why do we not validate CertificateRequest fields while creating? What if the customer does not fill in CSR and all necessary fields for its creation during the Kafka user creation?

Reply:
refactored